DemoApp/Sources/Components/AppState.swift

```diff
@@ -52,6 +56,15 @@ final class AppState: ObservableObject {
             didUpdate(voIPPushToken: voIPPushToken)
             deferSetDevice = false
             deferSetVoipDevice = false
+            activeCallCancellable?.cancel()
+            activeCallCancellable = nil
+            if let streamVideo {
+                activeCallCancellable = streamVideo
+                    .state
+                    .$activeCall
+                    .receive(on: DispatchQueue.main)
+                    .sink { [weak self] in self?.activeCall = $0 }
```

Comment:
`assign(to:)` instead of `sink` could be nicer to read but in that case we would need something like `assignWeakly(to)`

```swift
extension Publisher where Self.Failure == Never {
public func assignNoRetain<Root>(to keyPath: ReferenceWritableKeyPath<Root, Self.Output>, on object: Root) -> AnyCancellable where Root: AnyObject {
    sink { [weak object] (value) in
        object?[keyPath: keyPath] = value
    }
  }
}
```

Reply:
yeah assign isn't great because it holds a strong reference. Assign weak is a common publisher extension but we don't have it. I think it's alright to leave with it (especially as its in the DemoApp and we prefer to keep things as vanilla as possible)